invenio_records_permissions/policies/records.py

```diff
@@ -52,13 +53,13 @@ class RecordPermissionPolicy(BasePermissionPolicy):
     can_list = [AnyUser()]
     # Create action given to no one. Not even superusers.
     can_create = [Deny()]
-    # Read access given to everyone if public record/files and owners always.
-    can_read = [AnyUserIfPublic(), RecordOwners()]
+    # Read access given according to access rights, permissions and owners.
+    can_read = [AnyUserIfPublic(), AllowedIdentities('read'), RecordOwners()]
```

Comment:
Mayor: Again TBD, this should be configurable (same for all `AllowIdentitites`). I am not sure if both options should come out of the box, here we are mixing implicit and explicit generators.

I would create two policies, then if the user wants to mix them (at its own risk) they can implement its own policy.

Reply:
Not sure I understand this one.

- `AllowedIdentities()` is configurable (by passing 'read', 'update'...), no? Something else should be configurable?
- Mixing of implicit and explicit: yes mixing seems fine to me because both explicit and implicit generators define how to extract permissions from the metadata. The alternative I had in mind was to have the functionality of `AllowedIdentities` be baked into the `needs`, `excludes` methods of the Policy. This way custom Policies can't mistakenly forget to add this generator (but then they can't purposefully remove this generator either)

We can go over this together.

Reply:
TODO: Make it fully clear that the purpose of this permission is for additional permission